filebeat/cmd/root.go

```diff
@@ -25,5 +25,8 @@ func init() {
 	RootCmd.PersistentFlags().AddGoFlag(flag.CommandLine.Lookup("M"))
 	RootCmd.TestCmd.Flags().AddGoFlag(flag.CommandLine.Lookup("modules"))
 	RootCmd.SetupCmd.Flags().AddGoFlag(flag.CommandLine.Lookup("modules"))
-	RootCmd.AddCommand(cmd.GenModulesCmd(Name, "", buildModulesManager))
+
+	modulesCmd := cmd.GenModulesCmd(Name, "", buildModulesManager)
+	modulesCmd.AddCommand(genGenerateCmd())
```

Comment:
I would probably move this out of the `module` command do directly have `filebeat generate ...`. Like this we can use it to generate things which are not necessarly attached to module in the future.

Reply:
I think the current functionality is good under `modules`. But as we move away from `make` e.g. eliminating `make update`, I can totally see all generators under `generate` where it also includes generating of index templates, index patterns, etc.

From users point of view you are right about moving it to `generate`. So as we move more functionality here, users would not be confused if `modules generate` disappears and `generate` shows up instead.

Reply:
We should decide now in which package it belongs in the long run as moving it later would be a breaking change.

Reply:
Agreed